runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,31 @@
+local M = {}
+
+-- This function checks the first 15 lines for appearance of 'FoamFile'
+-- and then 'object' in a following line.
+-- In that case, it's probably an OpenFOAM file
+function M.foam()
+  local foam_file = false
+  for line = 1, 15 do
+    local current_line = vim.fn.getline(line)
```

Comment:
Use `vim.api.nvim_buf_get_lines(bufnr, 0, 15, true)` and then iterate over that with `ipairs()` instead of calling `vim.fn.getline` in each iteration. Also modify this function to accept a `bufnr` parameter and pass it in `filetype.lua`.

Reply:
probably best to copy the `getline` function from `filetype.lua`.